browser/src/actions/results.ts

```diff
@@ -20,247 +19,252 @@ export const notifyIsFetchingCommit = createAction<string>(Actions.IS_FETCHING_C
 export const fetchedAvatar = createAction<Avatar[]>(Actions.FETCHED_AVATARS);
 export const fetchedAuthors = createAction<ActionedUser[]>(Actions.FETCHED_AUTHORS);
 
-function getQueryUrl(store: RootState, baseUrl: string, args: string[] = []): string {
-    const id = store.settings.id || '';
-    const queryArgs = args.concat([`id=${encodeURIComponent(id)}`]);
-    const serverurl = window['server_url'];
-    return `${serverurl}${baseUrl}?${queryArgs.join('&')}`;
-}
-export const actionCommit = (logEntry: LogEntry, name: string = '', value: string = '') => {
-    // tslint:disable-next-line:no-any
-    return async (dispatch: Dispatch<any>, getState: () => RootState) => {
-        dispatch(notifyIsFetchingCommit(logEntry.hash.full));
-        const state = getState();
-        const url = getQueryUrl(state, `action/${name}`, [`value=${encodeURIComponent(value)}`]);
-        return axios.post(url, logEntry).then(result => {
-            switch (name) {
-                case 'reset_soft':
-                case 'reset_hard':
-                    dispatch(refresh());
-                    break;
-                case 'newtag':
-                    break;
-                case 'newbranch':
-                    dispatch(getBranches());    
-                    break;
-            }
-            dispatch(updateCommitInList(result.data as LogEntry));
-        });
+export namespace ResultActions {
+    export const commitsRendered = createAction<number>(Actions.COMMITS_RENDERED);
+
+    export const actionCommit = (logEntry: LogEntry, name: string = '', value: string = '') => {
+        // tslint:disable-next-line:no-any
+        return async (dispatch: Dispatch<any>, getState: () => RootState) => {
+            dispatch(notifyIsFetchingCommit(logEntry.hash.full));
+
+            const store = getState();
+
+            store.vscode.api.postMessage({
```

Comment:
Can we still not have the response coming back in here.
I.e. keep redux thunk.
I think thats better, i.e. make a request and get the response back.

Reply:
What we could do is:
* Create a class that has a method `postMessage`
* That will post the message to extension with a unique guid.
* It will listen to incoming messages from vsc, if message id === the guid of message that was sent, now we have the response.
* At this point we can invoke the callback

Reply:
U already have some of that code in `window.addEventListener('message', this.postMessageParser.bind(this));` in `PostMessageResult`.
This way we can still have code such as :
```return <messageBus>.post(url, logEntry).then(result => {```